src/Platform/Action.php

```diff
@@ -0,0 +1,278 @@
+<?php
+
+namespace Utopia\Platform;
+
+use Utopia\Validator;
+use Exception;
+
+abstract class Action
+{
+    /**
+     * Request method constants
+     */
+    public const REQUEST_METHOD_GET        = 'GET';
+    public const REQUEST_METHOD_POST       = 'POST';
+    public const REQUEST_METHOD_PUT        = 'PUT';
+    public const REQUEST_METHOD_PATCH      = 'PATCH';
+    public const REQUEST_METHOD_DELETE     = 'DELETE';
+    public const REQUEST_METHOD_OPTIONS    = 'OPTIONS';
+    public const REQUEST_METHOD_HEAD       = 'HEAD';
+
+    protected ?string $httpMethod = null;
```

Comment:
Would be great to have all HTTP methods grouped together. Another idea is to try and use them as Traits and then we can have a trait for each platform.